### PR TITLE
Update the CSCStubMatcher and CSCDigiMatcher under Validation/MuonCSCDigis 

### PR DESCRIPTION
**Pull request by tahuang1991**
#### PR description:

1. update the LCT match condition in CSCStubmatcher to make it configurable: either loose match or tight match
2. add number of digis counter to CSCDigiMatcher for multiplicity study

#### PR validation:

The validation is done by running GEMCSCAnalyzer under https://github.com/gem-sw/GEMCode/tree/for_CMSSW_12_5_0_x

### CODE REVIEW

**Comment by emanueleusai** [comment]
please test

**Comment by emanueleusai** [comment]
+1

**Review by perrotta — COMMENTED** [review]
Please rename the new getter methods without "get" (see naming rule 11 in http://cms-sw.github.io/cms_coding_rules.html)

File: Validation/MuonCSCDigis/interface/CSCDigiMatcher.h [hunk 1]
```diff
@@ -58,6 +58,11 @@ class CSCDigiMatcher {
   const CSCWireDigiContainer& wireDigisInDetId(unsigned int) const;
   const CSCWireDigiContainer& wireDigisInChamber(unsigned int) const;
 
+  //total digis, no match required
+  int getTotalComparators(unsigned int) const;
+  int getTotalStrips(unsigned int) const;
+  int getTotalWires(unsigned int) const;
```
[the remark this answers is not in the document]
Comment (tahuang1991) [reply Validation/MuonCSCDigis/interface/CSCDigiMatcher.h:64]: Sure. I will do it

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
please test

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**